taskiq/middlewares/opentelemetry_middleware.py

```diff
@@ -0,0 +1,299 @@
+import logging
+from contextlib import AbstractContextManager
+from typing import Any, Dict, Optional, Tuple, TypeVar
+
+try:
+    import opentelemetry  # noqa: F401
+except ImportError as exc:
+    raise ImportError(
+        "Cannot import opentelemetry_middleware. "
+        "Please install 'taskiq[opentelemetry]'.",
+    ) from exc
+
+from opentelemetry import context as context_api
+from opentelemetry import trace
+from opentelemetry.metrics import Meter, MeterProvider, get_meter
+from opentelemetry.propagate import extract, inject
+from opentelemetry.semconv.trace import SpanAttributes
+from opentelemetry.trace import Span, Tracer, TracerProvider
+from opentelemetry.trace.status import Status, StatusCode
+
+from taskiq import TaskiqMessage, TaskiqMiddleware, TaskiqResult, __version__
+
+logger = logging.getLogger("taskiq.opentelemetry")
+
+T = TypeVar("T")
+
+# Taskiq Context key
+CTX_KEY = "__otel_task_span"
+
+# Taskiq Context attributes
+TASKIQ_CONTEXT_ATTRIBUTES = [
+    "_retries",
+    "delay",
+    "max_retries",
+    "retry_on_error",
+    "timeout",
+    "X-Taskiq-requeue",
+]
+
+# Task operations
+_TASK_TAG_KEY = "taskiq.action"
+_TASK_SEND = "send"
+_TASK_EXECUTE = "execute"
+
+_TASK_RETRY_REASON_KEY = "taskiq.retry.reason"
+_TASK_NAME_KEY = "taskiq.task_name"
+
+
+def set_attributes_from_context(span: Span, context: Dict[str, Any]) -> None:
+    """Helper to extract meta values from a Taskiq Context."""
+    if not span.is_recording():
+        return
+
+    for key in TASKIQ_CONTEXT_ATTRIBUTES:
```

Comment:
Why do we track only particular labels? Let's just iterate over all available labels. 

Also, I think that we also want to extract args and kwargs to set it on the span. For better tracing.  Instead of passing labels, you can pass TaskiqMessage itself and get everthing from it.

Reply:
🤔 🤔 🤔 

Let's leave it as it is right now. We might reconsider it later.